File: 2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long sum = 0;
        for(int ele : chalk){
            sum+=ele;
        }
        long rem = k % sum;
        for(int i=0 ; i < chalk.length ;i++){
            rem -=chalk[i];
            if(rem <0 ){
                return i;
            }
        }
        return 0;
    }
}

File: 2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.kt
class Solution {
    fun chalkReplacer(chalk: IntArray, k: Int): Int {
        var sum:Long = 0
        for(ele in chalk){
            sum+=ele
        }
        var rem:Long = k%sum
        for(i in 0..chalk.size - 1 ){
            rem-=chalk[i]
            if(rem<0)
                return i
        }
        return 0
    }
}